src/librbd/cache/Block.h

```diff
@@ -0,0 +1,69 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CACHE_BLOCK_MAP
+#define CEPH_LIBRBD_CACHE_BLOCK_MAP
+
+#include "librbd/cache/Types.h" 
+#include "include/lru.h"
+#include <mutex>
+#include <unordered_map>
+
+namespace librbd {
+namespace cache {
+ 
+struct Entry : public LRUObject {
+  uint64_t on_disk_id;
+  Entry() : on_disk_id(0) {}
+};
+
+struct Block {
+  std::mutex lock;
+
+  uint64_t block;
+  Entry* entry;
+  uint8_t status;
+
+  void *tail_block_io_request; ///< used to track ios on this block
+  bool in_process;        ///< true if this block has been in processing by thread
+  Block( uint64_t block ) : block(block),
+      entry(nullptr), status(NOT_IN_CACHE),
+      tail_block_io_request(nullptr),
+      in_process(false) {}
+};
+
+class BlockMap {
+public:
+  //chendi: add an universal map to pre-allocate in memory data for each block
```

Comment:
I believe Ideally we should not leave name of some person in code.
Then we need to open new PR for removal.
Looks forgotten to delete.